src/main/java/org/jitsi/jigasi/SipCallTransformerMonitor.java

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright @ 2018 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.jitsi.jigasi;
+
+import net.java.sip.communicator.service.protocol.media.*;
+import net.java.sip.communicator.util.*;
+import org.jitsi.impl.neomedia.*;
+import org.jitsi.impl.neomedia.rtcp.*;
+import org.jitsi.impl.neomedia.transform.*;
+import org.jitsi.service.neomedia.*;
+import org.jitsi.util.*;
+import org.jitsi.utils.*;
+import org.jitsi.utils.concurrent.*;
+
+import java.util.*;
+
+/**
+ * Monitors outgoing RTCP and passes only those which are relevant for that
+ * direction. Filters outgoing RTCP.BYEs for ssrc that we haven't seen. Also
+ * monitors last seen media or rtcp and if one is missing we
+ * send hole punch packets. It can happen that the xmpp call is dropped and
+ * the sip call participant is waiting then there will be no media ot RTCP to
+ * keep alive the stream from the other side.
+ *
+ * @author Damian Minkov
+ */
+public class SipCallTransformerMonitor
+    extends SinglePacketTransformerAdapter
+    implements TransformEngine
+{
+    /**
+     * The logger.
+     */
+    private final static Logger logger = Logger.getLogger(
+        SipCallTransformerMonitor.class);
+
+    /**
+     * The RTCP transformer.
+     */
+    private final RTCPTransformer rtcpTransformer = new RTCPTransformer();
+
+    /**
+     * List of seen incoming or outgoing ssrcs.
+     */
+    private Set<Long> seenSSRCs = new HashSet<>();
```

Comment:
The transformer chain is run in different threads, so this needs to be thread safe. We can use `ConcurrentHashMap.newKeySet()` for example